File: app/src/main/java/sixue/naivereader/provider/NetProvider.java
package sixue.naivereader.provider;

import android.content.Context;

import java.util.List;

import sixue.naivereader.data.Book;
import sixue.naivereader.data.Chapter;

public abstract class NetProvider {
    private boolean active = true;

    public abstract String getProviderId();

    public abstract String getProviderName();

    public abstract List<Book> search(String s, Context context);

    public abstract void downloadContent(Book book, String bookSavePath);

    public abstract void downloadChapter(Book book, Chapter chapter);

    public boolean isActive() {
        return active;
    }

    public void setActive(boolean active) {
        this.active = active;
    }
}

File: app/src/main/java/sixue/naivereader/provider/NetProvider.kt
package sixue.naivereader.provider

import android.content.Context
import sixue.naivereader.data.Book
import sixue.naivereader.data.Chapter

abstract class NetProvider {
    var isActive = true
    abstract val providerId: String
    abstract val providerName: String
    abstract fun search(s: String, context: Context): List<Book>
    abstract fun downloadContent(book: Book, bookSavePath: String): List<Chapter>
    abstract fun downloadChapter(book: Book, chapter: Chapter)
    abstract fun getChapterUrl(book: Book, chapter: Chapter): String
}